Added ipcalc.el

dotemacs opened the pull request:
### Brief summary of what the package does

IP calculator in Emacs Lisp.

When given an IP and subnet e.g. **192.168.0.23/21** it produces:

```
Address:   192.168.0.23         11000000101010000000000000010111
Netmask:   255.255.248.0 = 21   11111111111111111111100000000000
Wildcard:  0.0.7.255            00000000000000000000011111111111
=>
Network:   192.168.0.0          11000000101010000000000000000000
HostMin:   192.168.0.1          11000000101010000000000000000001
HostMax:   192.168.7.254        11000000101010000000011111111110
Broadcast: 192.168.7.255        11000000101010000000011111111111
Hosts/Net: 2046
```

### Direct link to the package repository

https://github.com/dotemacs/ipcalc.el

### Your association with the package

I'm the author and the maintainer.

### Relevant communications with the upstream package maintainer

**None**

### Checklist

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've built and installed the package using the instructions in the [README](https://github.com/melpa/melpa/blob/master/README.md)

Added ipcalc.el package recipe

syohex commented:
I have sent PR(https://github.com/dotemacs/ipcalc.el/pull/5)

And
- Please use `-` separator(`--` separator can be used for private functions/variables) for [Emacs Lisp Coding convention](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html#Coding-Conventions)
- Please set package prefix for all global functions and variables. Please don't define like `list-to-string`
- Should `ipcalc` function be defined as interactive command ?

dotemacs commented:
Hello @syohex 

Thanks for the feedback.

> I have sent PR(dotemacs/ipcalc.el#5)

Merged, thanks.

> Please use - separator(-- separator can be used for private functions/variables) for Emacs Lisp Coding convention

Done, see: https://github.com/dotemacs/ipcalc.el/commit/f4ac17eb8d4cb311b1419d89fe745e23881a19e2

> Please set package prefix for all global functions and variables. Please don't define like list-to-string

Done, removed: https://github.com/dotemacs/ipcalc.el/commit/8cbc6a46bdcd2c7f6856b4337f7c47e214fe266b

> Should ipcalc function be defined as interactive command ?

Yes, my bad, added the interactive form:
https://github.com/dotemacs/ipcalc.el/commit/be6fba84e4ce7074327b71e860d89c150c971e5e

Thanks for reviewing, is there anything else?

purcell commented:
Thanks!
